rspotify-model/src/album.rs

```diff
@@ -33,9 +32,6 @@ pub struct SimplifiedAlbum {
     pub release_date_precision: Option<String>,
     #[serde(skip_serializing_if = "Option::is_none")]
     pub restrictions: Option<Restriction>,
-    #[serde(rename = "type")]
-    pub _type: Type,
-    pub uri: Option<String>,
```

Comment:
It seems you just removed these `_type`, `uri` and `id` fields, could they serialize/deserialize into a single `AlbumId` type?

Same for the `FullAlbum` type

Reply:
My bad, I shouldn't have removed the `id` field.

Reply:
What do you mean? That I should remove the `Option` for the `Id` field?

Reply:
Yes, there are `Id` fields in both of these structs, then what I mean is that there are three fields named `_type`, `uri`, `id` in the `SimplifiedAlbum` struct, could they map into one individual field named `id`?

For example, there is a `Json` data like this:

```json
// `SimplifiedAlbum` data
{
 ...
 "id": "v5eMCfY3fg3kDdBnfD80ZOW9eYybjcND",
 "_type": "album",
 "uri":"spotify_uri_xxx"
 ...
}
```

And you want to map the above data to a `AlbumId` type, while we have three input fields, but have only one output field, it's many-to-one mapping:

`id`+`_type`+`uri` => `AlbumId`

As `serde` could only handle one-to-one mapping(As far as I know)

`id` => `AlbumId`, `_type` => `AlbumId` or `uri` => `AlbumId`.

It seems impossible to implement this serialization, perhaps you could write a unit test to check if it works.

Do I demonstrate this clear? Or I just misunderstand something about how `serde` works?

Reply:
You take a look at the `serde` document, probably the `flatten` attribute is what you want: https://serde.rs/attr-flatten.html

Reply:
The thing is that `_type` and `uri` are redundant and don't offer any more information than what we already have when deserializing. Given an ID we already know its type. So there's no need to even check these fields, we can just ignore them.

Reply:
I misunderstood your point, you are right.